devtools/src/panel/components/editable-json/JsonItem.vue

```diff
@@ -0,0 +1,93 @@
+<script setup lang="ts">
+import { ref, computed } from 'vue'
+import { isObject } from '@vueuse/core'
+import type { JSONValue } from '../../utils/json'
+import { formatValue, getValueTypeClass } from '../../utils/json'
+
+const props = defineProps<{
+  itemKey: string
+  value: JSONValue
+  depth: number
+}>()
+
+const isExpanded = ref(false)
+const isHovered = ref(false)
+
+const isExpandable = computed(() => {
+  return (
+    (Array.isArray(props.value) && props.value.length > 0) ||
+    (isObject(props.value) && Object.keys(props.value).length > 0))
+})
+
+function toggleExpansion() {
+  isExpanded.value = !isExpanded.value
+}
+
+function handleMouseEnter() {
+  isHovered.value = true
+}
+
+function handleMouseLeave() {
+  isHovered.value = false
+}
+
+ function getCollectionLabel(value: unknown): string {
+  if (Array.isArray(value)) {
+    return `Array[${value.length}]`
+  }
+  if (isObject(value) && value !== null) {
+    return 'Object'
+  }
+  return ''
+}
+</script>
+
+<template>
+  <div
+    class="text-sm mb-1"
+    :style="{ paddingLeft: `${depth * 20}px` }"
+    @mouseenter="handleMouseEnter"
+    @mouseleave="handleMouseLeave"
+  >
+    <!-- Expandable items (Arrays and Objects) -->
+    <div v-if="isExpandable" class="px-3">
```

Comment:
I think we could refactor this code for the expandable bit to only affect the specific parts (chevron and collection label). It should make it easier to apply a consistent style to primitives and expandable values